content/chainguard/chainguard-enforce/chainguard-enforce-signing/getting-started-chainguard-enforce-signing.md

```diff
@@ -0,0 +1,213 @@
+
+We will publish this container to [ttl.sh](https://ttl.sh/), which is an anonymous and ephemeral container registry. If you would prefer to publish this in another container registry, feel free to do so. 
+
+We'll start by building the image. The `1h` tag tells ttl.sh to store the image in their registry for one hour, you can modify the time for your purposes.
+
+```sh
+sudo docker build -t ttl.sh/$USERNAME/hello-signing:1h .
+```
+
+Next, push the image to the registry.
+
+```sh
+docker push ttl.sh/$USERNAME/hello-signing:1h
+```
+
+At this point, you will receive output that your container was published and you are ready to continue to the final step.
+
+## Step 5 — Signing and Verifying an Image
+
+You are now ready to sign and verify an image with your brand new Certificate Authority! 
+
+First, [log into Chainguard](/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/log-in-chainguard-enforce/#signing-in-through-chainctl):
+
+```sh
+chainctl auth login
+```
+
+This workflow will ask you to go through an [OIDC flow](software-security/glossary/#oidc) to authenticate with Chainguard.
+
+Next, you’ll need to set up your environment to point Cosign to your personal Enforce Signing instance. Here, replace `$ENFORCE_CA_NAME` with the name of your Enforce CA. You can pull this up again by running `chainctl sigstore ca ls `. 
+
+```sh
+eval $(chainctl sigstore env $ENFORCE_CA_NAME)
+```
+
+You will have to authenticate again, to get an authentication token unique to your instance of Enforce Signing. This command will set the required environment variables for Cosign to sign your image with your private CA. 
+
+You are now ready to sign by following the steps below. If you followed [Step 4](/chainguard/chainguard-enforce/chainguard-enforce-signing/getting-started-chainguard-enforce-signing/#step-4--set-up-and-publish-a-container-image), replace `$IMAGE` with `ttl.sh/$USERNAME/hello-signing:1h`, otherwise fill in your own image that you would like to sign. 
+
+```sh
+eval $(chainctl sigstore env $ENFORCE_CA_NAME)
+cosign sign $IMAGE
+```
+
+Finally, you are ready to verify your image. The `cosign verify` command expects both a certificate identity and an issuer. The issuer, `https://auth.chainguard.dev`, is already set by the `sigstore env` command. The certificate identity will correspond to the email address you used to authenticate to Chainguard. Replace `$EMAIL` below with that email address.
+
+```sh
+eval $(chainctl sigstore env $ENFORCE_CA_NAME)
+cosign verify $IMAGE --certificate-identity=$EMAIL
+```
+
+You can also verify with a wildcard certificate identity, as in the following command.
+
+```sh
+eval $(chainctl sigstore env $ENFORCE_CA_NAME)
+cosign verify $IMAGE --certificate-identity-regexp=".*"
+```
+
+After running this command, you should receive output that returns that the Cosign claims and certificate were verified.
```

Comment:
Is it the same / similar output for both of these `cosign verify` commands? This output looks like what I received from running the first command, but as I mentioned the second one didn't work for me.

Reply:
It should be the same output either way